pandas_transform/pandas_transform.md

```diff
@@ -0,0 +1,990 @@
+</script>
+</lia-keep>
+</div>
+
+That code didn't have any output because we didn't ask it to print anything, but it did create the `covid_testing` DataFrame. We can take a look at with `print(covid_testing)`:
+
+<div class="python_data">
+<lia-keep>
+<script type="text/x-sage">
+print(covid_testing)
+</script>
+</lia-keep>
+</div>
+
+Unless you have an extremely wide browser window, the output above is broken into as many columns as it can fit at a time. Some interfaces will output the DataFrame and let you scroll right to see more columns, but this interface just prints as many columns as it can fit and then starts again with the next columns. Good thing it didnt't print all 15524 rows!
+
+When you print a DataFrame or Series you will see the first five rows and the last five rows of a DataFrame. If, like this one, it has more than 10 rows, the hidden rows will be indicated by ellipses. You can also ask for the first or last 5 rows of data using the methods `.head()` and `.tail()`. Try putting a number in the parentheses to get different numbers of rows.
+
+<div class="python_data">
+<lia-keep>
+<script type="text/x-sage">
+print(covid_testing.head())
+</script>
+</lia-keep>
+</div>
+<br>
+
+**Loading your own data**
+
+To create a DataFrame of your data, make sure you use the right command for your file type, and make sure the location of the file is in quotes.
+| File type | Read command |
+| - | - |
+| .csv | pd.read_csv('location')|
+| .xlsx | pd.read_excel('location')|
+| .ods | pd.read_excel('location')|
+| .json| pd.read_json('location')|
+| .html| pd.read_html('location')|
+| .sql | pd.read_sql('location')|
+
+
+
+### Column names and row indices
+@sage
+From now on we will add the definition of our DataFrame `covid_testing` to the set-up of each page. There is nothing for you to click here, this code ran automatically when you loaded the page:
+
+<div class="python_data_init">
+<lia-keep>
+<script type="text/x-sage">
+import pandas as pd
+covid_testing = pd.read_csv('https://raw.githubusercontent.com/arcus/education_modules/embedded_code/a_sample_module_template/covid_testing.csv')
+</script>
+</lia-keep>
+</div>
+
+When a DataFrame is displayed it has a row of column names along the top, and a column of indices along the left side. In some environments, like Jupyter notebooks, these names and indices may be bolded.
+
+The methods `.columns` and `.index` will show you all of the column and row names, respectively. **Give it a try:**
+
+<div class="python_data">
+<lia-keep>
+<script type="text/x-sage">
+print(covid_testing.columns)
+</script>
+</lia-keep>
+</div>
+
+The column names were imported with the `.csv` file. Since the original file didn't have labels for the rows, `pandas` automatically labeled them numerically 0 through 15523. If you imported tabular data that didn't have column headers, the column names would be numeric, the way the row indices are in our `covid_testing` DataFrame.
```

Comment:
When I run `print(covid_testing.index)`, it looks like the range is 0 to 15524, not 15523. Or does python give ranges that are not inclusive on the high end, perhaps? If so, worth a mention, since right now it looks like a typo.

Reply:
Good call! I added a help box for python ranges because they are non-intuitive and occasionally infuriating!